### DISCUSSION TITLE
Feature Request for Batch Editor

### DISCUSSION THREAD

**Discussion by ReZeroEmiria** [kwsch/PKHeX#3651] [Q&A]
I'd like to request a new command/tag for use in the Batch editor.

Currently we can use 

.PID=$rand
.PID=$shiny0
.PID=$shiny1

To reroll these values and randomize pokemon. I'd like to suggest new values of "$randshiny1" and "$randshiny0" so that it rerolls the value even if the pokemon is already shiny0 or shiny1, and rerolls it into the appropriate shiny type specified.

Currently if you have a box of shiny0 (square shiny) pokemon and use "$shiny0" it detects all pokemon as already fulfilling the requirement and does not reroll the value. 

I would like to be able to reroll PID of a shiny into another valid PID. Hope this can be done! Thank you developers for your work!

Not sure if intentional as well but I cannot find the value for hatch counter/ happiness in the batch editor for these new pokemon. Would like to be able to mass edit eggs to hatch quickly as well.

**Comment by kwsch** [comment]
```
=ShinyXor=0
.PID=$rand
.PID=$shiny0
```
Above commands will filter to only square shinies (Xor = 0), then will randomize their PID, then will make them Xor0 square again.

You can do the same for `=ShinyXor=1`, or even `=Shiny=true`.

Hatch Counter uses the same field as the OT Friendship.

**Comment by ReZeroEmiria** [comment]
Did some research on Dudunsparce/Maushold Evolution

Would also like to suggest something akin to

.EncryptionConstant=$randlucky

where it will roll the Encryption constant so EC mod 100 == 0
This can also implemented via some sort of command key + clicking EC Reroll button under OT/Misc 


Thank you for clarifying about OT friendship, I feel it was named something different before, but I see the field now.
Also was about to suggest something for EC reroll, unless there is already a way to implement this. Thank you for taking the time to review this!

**Comment by kwsch** [comment]
Added batch command in the latest commit - https://github.com/kwsch/PKHeX/commit/731a3fdc3be5208dfe97dab163e307f90e140d8f

For the main GUI, it's probably easier to just require users to use a calculator instead of hiding it behind some quirky key combination that only applies to those species. Same as it has been for those wanting control over the EC characteristic tiebreaker -- just come up with a single value yourself by math-ing it out :)
